rd_stats.c

```diff
@@ -1826,8 +1826,14 @@ void read_cpuinfo(struct stats_pwr_cpufreq *st_pwr_cpufreq, int nbr)
 			sscanf(strchr(line, ':') + 1, "%u", &proc_nb);
 		}
 
-		else if (!strncmp(line, "cpu MHz\t", 8)) {
-			sscanf(strchr(line, ':') + 1, "%u.%u", &ifreq, &dfreq);
+		else if (!strncmp(line, "cpu MHz\t", 8) ||
+			 !strncmp(line, "clock\t", 6)) {
+
+			if (strstr(line, "MHz"))
```

Comment:
This condition is always true ("MHZ" exists for both architectures). I removed it to keep only the line below:
``
sscanf(strchr(line, ':') + 1, "%u.%u", &ifreq, &dfreq);
``
which works whether the clock rate is followed by MHz or not.